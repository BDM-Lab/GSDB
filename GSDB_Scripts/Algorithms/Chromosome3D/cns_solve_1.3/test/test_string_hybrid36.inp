! Test Hybrid-36 encode and decode
evaluate ( $1 = decode("9999"))
evaluate ( $1 = decode("A000"))
evaluate ( $1 = decode("ZZZZ"))
evaluate ( $1 = decode("a000"))
evaluate ( $1 = decode("zzzz"))
! CAUTION: hybrid-36 accepts a blank character as zero (as implemented
! in the CCTBX hybrid_36_f.f routines)
evaluate ( $1 = decode("a  0"))
evaluate ( $1 = decode("a   "))
! The next two statements should fail; only 4-char hybrid36 is supported.
evaluate ( $1 = decode("A0000"))
evaluate ( $1 = decode("A00"))

evaluate ( $1 = format("I4",9999))    ! A000 - 1
evaluate ( $1 = format("I4",10000))   ! A000
evaluate ( $1 = format("I4",1223055)) ! ZZZZ
evaluate ( $1 = format("I4",1223056)) ! a000
evaluate ( $1 = format("I4",2436111)) ! zzzz

! For the near future, use a fixed-format and trim leading spaces
! using the new string functions:
evaluate ( $1 = trim(adjustl(format("I20",1223056))))

stop

