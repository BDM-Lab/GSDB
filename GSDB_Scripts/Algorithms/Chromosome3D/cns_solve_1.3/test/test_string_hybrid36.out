          ============================================================
          |                                                          |
          |            Crystallography & NMR System (CNS)            |
          |                         CNSsolve                         |
          |                                                          |
          ============================================================
           Version: 1.3
           Status: General release
          ============================================================
           Written by: A.T.Brunger, P.D.Adams, G.M.Clore, W.L.DeLano,
                       P.Gros, R.W.Grosse-Kunstleve,J.-S.Jiang,J.M.Krahn,
                       J.Kuszewski, M.Nilges, N.S.Pannu, R.J.Read,
                       L.M.Rice, G.F.Schroeder, T.Simonson, G.L.Warren.
           Copyright (c) 1997-2010 Yale University
          ============================================================
           Running on machine: lewis4-r710-login-node223 (x86_64/Linux,64-bit)
                               with 16 threads
           Program started by: oeow39
           Program started at: 16:08:09 on 05-Jul-2018
          ============================================================

 FFT3C: Using FFTPACK4.1

 CNSsolve>! Test Hybrid-36 encode and decode 
 CNSsolve>evaluate ( $1 = decode("9999")) 
 EVALUATE: symbol $1 set to    9999.00     (real)
 CNSsolve>evaluate ( $1 = decode("A000")) 
 EVALUATE: symbol $1 set to    10000.0     (real)
 CNSsolve>evaluate ( $1 = decode("ZZZZ")) 
 EVALUATE: symbol $1 set to   0.122306E+07 (real)
 CNSsolve>evaluate ( $1 = decode("a000")) 
 EVALUATE: symbol $1 set to   0.122306E+07 (real)
 CNSsolve>evaluate ( $1 = decode("zzzz")) 
 EVALUATE: symbol $1 set to   0.243611E+07 (real)
 CNSsolve>! CAUTION: hybrid-36 accepts a blank character as zero (as implemented 
 CNSsolve>! in the CCTBX hybrid_36_f.f routines) 
 CNSsolve>evaluate ( $1 = decode("a  0")) 
 EVALUATE: symbol $1 set to   0.122306E+07 (real)
 CNSsolve>evaluate ( $1 = decode("a   ")) 
 EVALUATE: symbol $1 set to   0.122306E+07 (real)
 CNSsolve>! The next two statements should fail; only 4-char hybrid36 is supported. 
 CNSsolve>evaluate ( $1 = decode("A0000")) 
 %XRDECODE-ERR: Illegal argument(s).:
 evaluate ( $1 = decode("A0000")) 
                                ^
 %EVALUATE-ERR: EVALUATE assignment aborted.:
 evaluate ( $1 = decode("A0000")) 
                                ^
 CNSsolve>evaluate ( $1 = decode("A00")) 
 %XRDECODE-ERR: Illegal argument(s).:
 evaluate ( $1 = decode("A00")) 
                              ^
 %EVALUATE-ERR: EVALUATE assignment aborted.:
 evaluate ( $1 = decode("A00")) 
                              ^
 CNSsolve> 
 CNSsolve>evaluate ( $1 = format("I4",9999))    ! A000 - 1 
 EVALUATE: symbol $1 set to "9999" (string)
 CNSsolve>evaluate ( $1 = format("I4",10000))   ! A000 
 EVALUATE: symbol $1 set to "A000" (string)
 CNSsolve>evaluate ( $1 = format("I4",1223055)) ! ZZZZ 
 EVALUATE: symbol $1 set to "ZZZZ" (string)
 CNSsolve>evaluate ( $1 = format("I4",1223056)) ! a000 
 EVALUATE: symbol $1 set to "a000" (string)
 CNSsolve>evaluate ( $1 = format("I4",2436111)) ! zzzz 
 EVALUATE: symbol $1 set to "zzzz" (string)
 CNSsolve> 
 CNSsolve>! For the near future, use a fixed-format and trim leading spaces 
 CNSsolve>! using the new string functions: 
 CNSsolve>evaluate ( $1 = trim(adjustl(format("I20",1223056)))) 
 EVALUATE: symbol $1 set to "1223056" (string)
 CNSsolve> 
 CNSsolve>stop 
          ============================================================
           Maximum dynamic memory allocation:     8201488 bytes
           Maximum dynamic memory overhead:           208 bytes
           Program started at: 16:08:09 on 05-Jul-2018
           Program stopped at: 16:08:09 on 05-Jul-2018
           CPU time used:       0.0036 seconds
          ============================================================
